nvflare/private/fed/server/job_cmds.py

```diff
@@ -422,6 +435,34 @@ def get_job_meta(self, conn: Connection, args: List[str]):
                     f"job {job_id} does not exist", meta=make_meta(MetaStatusValue.INVALID_JOB_ID, job_id)
                 )
 
+    def list_job_components(self, conn: Connection, args: List[str]):
+        if len(args) < 2:
+            conn.append_error("Usage: list_job_components job_id", meta=make_meta(MetaStatusValue.SYNTAX_ERROR))
+            return
+
+        job_id = conn.get_prop(self.JOB_ID)
+        engine = conn.app_ctx
+        job_def_manager = engine.job_def_manager
+        if not isinstance(job_def_manager, JobDefManagerSpec):
+            raise TypeError(
+                f"job_def_manager in engine is not of type JobDefManagerSpec, but got {type(job_def_manager)}"
+            )
+        with engine.new_context() as fl_ctx:
+            list_of_data = job_def_manager.list_client_data(jid=job_id, fl_ctx=fl_ctx)
+            if list_of_data:
+                filtered_data = [
+                    item for item in list_of_data if item not in {"workspace", "meta", "scheduled", "data"}
+                ]
```

Comment:
Or another way we move these filter mechanism into the "list_client_data" method, so that it does filter out the server-generated data?

One question here is that how do we generate these keys ("meta", "scheduled", "data"), do we have any constant value we can use?

Reply:
These keys are special since we use them in the system ourselves. For now, I put it into a variable system_components to make it clearer.